samples/nrf_desktop/src/hw_interface/buttons_matrix.c

```diff
@@ -191,17 +294,26 @@ void button_pressed(struct device *gpio_dev, struct gpio_callback *cb,
 		}
 	}
 
-	if (!atomic_get(&active)) {
+	switch (state) {
+	case STATE_IDLE:;
 		struct wake_up_event *event = new_wake_up_event();
 		EVENT_SUBMIT(event);
-
-		return;
+		break;
+
+	case STATE_ACTIVE:
+		state = STATE_SCANNING;
+		k_delayed_work_submit(&matrix_scan, 0);
+		break;
+
+	case STATE_SCANNING:
```

Comment:
In this callback STATE_SCANNING is forbidden, so please add `/* Fallthough */` comment as usually in cases without `break`.

Reply:
I would prefer not. It's easier to read this way as cases going this way are next to each other. I use fall through when case that does something do not end with break (intentionally shares part of code with case directly below).
Note that we have the same style already in out code.